torch/distributed/_tensor/placement_types.py

```diff
@@ -51,13 +51,6 @@ def _split_tensor(
             few ranks before calling the collectives (i.e. scatter/all_gather, etc.).
             This is because collectives usually require equal size tensor inputs
         """
-        assert (
```

Comment:
hmmm why we need to this dim < ndim assertion? I think this assertion is still valid?